src/debugger/gdb-stub.c

```diff
@@ -433,7 +453,35 @@ static void _processQSupportedCommand(struct GDBStub* stub, const char* message)
 		}
 		message = end + 1;
 	}
-	strncpy(stub->outgoing, "swbreak+;hwbreak+", GDB_STUB_MAX_LINE - 4);
+	strncpy(stub->outgoing, "swbreak+;hwbreak+;qXfer:features:read+;qXfer:memory-map:read+", GDB_STUB_MAX_LINE - 4);
+}
+
+static void _processQXferCommand(struct GDBStub* stub, const char* params, const char* data) {
+	unsigned int offset = 0;
+	unsigned int length = 0;
+
+	unsigned int index = 0;
+	while (params[index] != ',') {
+		offset <<= 4;
+		offset |= _hex2int(&params[index], 1);
+		index++;
+	}
+
+	index++;
+	while (index < strlen(params) - 3) {
+		length <<= 4;
+		length |= _hex2int(&params[index], 1);
+		index++;
+	}
```

Comment:
```suggestion
	++index;
	for (; params[index] != ','; ++index) {
		offset <<= 4;
		offset |= _hex2int(&params[index], 1);
	}
```

Reply:
Just to confirm: you just want to convert the while to a for? because your suggestion changes the check in a way that might not work (there is only one comma, the length - 3 is to ignore the checksum so this just checks if it's the end of the packet minus the checksum part).

Reply:
Sorry, that was just me being blind and copy/pasting. It should be fixed now.